js/src/common/Application.tsx

```diff
@@ -287,11 +328,13 @@ export default class Application {
    * Make an AJAX request, handling any low-level errors that may occur.
    *
    * @see https://mithril.js.org/request.html
-   * @param {Object} options
+   *
+   * @param options
    * @return {Promise}
-   * @public
    */
-  request(originalOptions) {
+  request<ResponseType>(
+    originalOptions: Mithril.RequestOptions<ResponseType> & { errorHandler: (errorMessage: string) => void; url: string }
```

Comment:
Complex one-liner types are hard to parse: can we declare these up top and/or split these into a composition of simpler type declarations?